File: Agenda_VB/Agenda_VB/Persona.vb
Public Class Persona
    Protected nombre As String
    Protected fechaNacimiento As DateTime?
    Public Property Nombr() As String
        Get
            Return nombre
        End Get
        Set(ByVal Value As String)
            nombre = Value
        End Set
    End Property
    Public Property FechaNacimient() As DateTime?
        Get
            Return fechaNacimiento
        End Get
        Set(ByVal Value As DateTime?)
            fechaNacimiento = Value
        End Set
    End Property
    Public ReadOnly Property Eda() As Integer
        Get
            Dim edad As Integer
            edad = (DateTime.Now.Year - fechaNacimiento.Value.Year)
            Return edad
        End Get
    End Property
    Private Function Persona()
        nombre = String.Empty
        fechaNacimiento = Nothing 'DateTime.MinValue
    End Function
    Private Function Persona(ByVal nombre As String, ByVal fechaNacimiento As DateTime?)
        Me.nombre = nombre
        Me.fechaNacimiento = fechaNacimiento
    End Function
    Public Overrides Function ToString() As String
        Return Nombr.ToUpper + " " + Convert.ToString(Eda)
    End Function
End Class

File: Agenda_C/Agenda_C/Persona.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Agenda_C
{
    internal class Persona
    {
		protected String nombre;
		public String Nombre
		{
			get { return nombre; }
			set { nombre = value; }
		}
		protected DateTime? fecha;

		public DateTime? Fecha
		{
			get { return fecha; }
			set { fecha = value; }
		}

		public int Edad
		{
			get 
			{
				int edad = (DateTime.Now.Year - fecha.Value.Year);
				return edad;
			}
		}
        public Persona()
        {
            nombre = String.Empty;
            fecha = null;
        }
        public Persona(String nombre, DateTime? fecha)
        {
            this.nombre = nombre;
            this.fecha = fecha;
        }

        public override string ToString()
        {
            return nombre.ToUpper() + " " + Edad;
        }
    }
}
